packages/babel-plugin-extract/src/index.tsx

```diff
@@ -0,0 +1,198 @@
+import { declare } from '@babel/helper-plugin-utils';
+import * as t from '@babel/types';
+import { NodePath } from '@babel/core';
+
+const isCreateElement = (node: t.Node): node is t.CallExpression => {
+  return (
+    t.isMemberExpression(node) &&
+    t.isIdentifier(node.object) &&
+    node.object.name === 'React' &&
+    t.isIdentifier(node.property) &&
+    node.property.name === 'createElement'
+  );
+};
+
+const isJsxRuntime = (node: t.Node): node is t.CallExpression => {
+  return t.isCallExpression(node) && t.isIdentifier(node.callee) && node.callee.name === '_jsx';
+};
+
+const isJsxsRuntime = (node: t.Node): node is t.CallExpression => {
+  return t.isCallExpression(node) && t.isIdentifier(node.callee) && node.callee.name === '_jsxs';
+};
+
+const getJsxRuntimeChildren = (node: t.CallExpression): Array<t.Expression> => {
+  const props = node.arguments[1];
+  const children: t.Expression[] = [];
+
+  if (t.isObjectExpression(props)) {
+    props.properties.forEach((prop) => {
+      if (t.isObjectProperty(prop) && t.isExpression(prop.value)) {
+        children.push(prop.value);
+      }
+    });
+  }
+
+  return children;
+};
+
+const removeStyleDeclarations = (node: t.Node, parentPath: NodePath<any>, pass: PluginPass) => {
```

Comment:
```suggestion
const removeStyleDeclarations = <T extends {} = any>(node: t.Node, parentPath: NodePath<T>, pass: PluginPass) => {
```
Dunno, may get better type inference from consumers?